sys-libs/mesa/mesa-18.2.0-rc2.recipe

```diff
@@ -0,0 +1,149 @@
+SUMMARY="Multi-platform GL implementation"
+DESCRIPTION="Mesa is an open-source implementation of the OpenGL \
+specification. The OpenGL specification documents a system for rendering \
+interactive 3D graphics. Mesa fills the role of the Haiku OpenGL kit \
+providing 3D rendering to Haiku applications."
+HOMEPAGE="https://www.mesa3d.org/"
+COPYRIGHT="1999-2018 Brian Paul"
+LICENSE="MIT"
+REVISION="1"
+SOURCE_URI="https://mesa.freedesktop.org/archive/mesa-${portVersion}.tar.xz"
+CHECKSUM_SHA256="2201d99bd0e32f1f36f1714a11d8ad270e4f560488b77acdb811d083e6157a4a"
+PATCHES="mesa-$portVersion.patchset"
+
+ARCHITECTURES="!x86_gcc2 x86 x86_64 ?arm ?ppc"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+ mesa$secondaryArchSuffix = $portVersion
+ lib:libGL$secondaryArchSuffix = 1.0.0 compat >= 1
+ lib:libEGL$secondaryArchSuffix = 1.0.0 compat >= 1
+ lib:libOSMesa$secondaryArchSuffix = 8.0.0 compat >= 8
+ "
+REQUIRES="
+ haiku$secondaryArchSuffix
+ lib:libz$secondaryArchSuffix
+ "
+
+PROVIDES_devel="
+ mesa${secondaryArchSuffix}_devel = $portVersion
+ devel:libGL$secondaryArchSuffix = 1.0.0 compat >= 1
+ devel:libEGL$secondaryArchSuffix = 1.0.0 compat >= 1
+ devel:libOSMesa$secondaryArchSuffix = 8.0.0 compat >= 8
+ "
+REQUIRES_devel="
+ mesa$secondaryArchSuffix == $portVersion base
+ "
+
+SUMMARY_swpipe="The Mesa LLVM enhanced Gallium software pipe renderer"
+DESCRIPTION_swpipe="This 3D BGLRenderer add-on provides Gallium LLVM \
+enhanced software rendering. Software pipe rendering performs all \
+3D rendering on the systems CPU and doesn't require any specialized \
+hardware. The usage of LLVM over traditional rasterization gives this \
+renderer a boost in performance.
+
+Gallium software pipe rendering is in an extremely early state as of \
+this version of Mesa, and may not function as expected."
+
+PROVIDES_swpipe="
+ mesa${secondaryArchSuffix}_swpipe = $portVersion
+ "
+REQUIRES_swpipe="
+ mesa$secondaryArchSuffix == $portVersion base
+ lib:libLLVM_5.0$secondaryArchSuffix
+ "
+
+BUILD_REQUIRES="
+ haiku${secondaryArchSuffix}_devel
+ devel:libz$secondaryArchSuffix
+ devel:libLLVM_5.0$secondaryArchSuffix
+ "
+BUILD_PREREQUIRES="
+ cmd:aclocal
+ cmd:autoconf
+ cmd:automake
+ cmd:llvm_config
+ cmd:egrep
+ cmd:gcc$secondaryArchSuffix
+ cmd:ld$secondaryArchSuffix
+ cmd:libtoolize$secondaryArchSuffix
+ cmd:make
+ cmd:pkg_config$secondaryArchSuffix
+ cmd:which
+ "
+
+BUILD()
+{
+ if [ -n "$secondaryArchSuffix" ]; then
+ export HAIKU_SECONDARY_ARCH="$effectiveTargetArchitecture"
+ fi
+
+ autoreconf -fi
+ # for whatever reason, the llvm_require_version function is busted, so we
+ # manually patch it here
+ sed -i 's@"\$LLVM_VERSION_MAJOR" -gt "\$llvm_target_version_major"@true@' configure
+ sed -i 's@-DHAVE_LLVM=0x0\$LLVM_VERSION_INT@-DHAVE_LLVM=0x0500@' configure
+ runConfigure ./configure \
```

Comment:
Please, do not remove the identation. Just the superfluous whitespaces should be removed at the end  of the lines